dist/SwupJsPlugin.js

```diff
@@ -162,6 +162,11 @@ var JsPlugin = function (_Plugin) {
 			var currentTransitionRoutes = _this.swup.transition;
 			var animation = _this.options[index];
 
+			if (!(animation && animation[type])) {
+				console.warn('No animation found');
+				return Promise.resolve();
+			}
+
```

Comment:
The code looks good, although I would probably prefer `!animation || !animation[type]`. I just tried to reproduce the error described in #3 , but I couldn't, so probably this was fixed somewhere else in swup's core, already?

Reply:
...still, probably a good thing to check for a valid animation, anyways. Will approve...